src/consensus/params.h

```diff
@@ -70,8 +72,13 @@ struct BIP9Deployment {
 struct Params {
     uint256 hashGenesisBlock;
     int nSubsidyHalvingInterval;
-    /* Block hash that is excepted from BIP16 enforcement */
-    uint256 BIP16Exception;
+    /**
+     * Hashes of blocks that
+     * - are known to be consensus valid, and
+     * - buried in the chain, and
+     * - fail if the default script verify flags are applied.
```

Comment:
That'd be ambiguous. You'd also have to specify that it means the flags that *this version* (the version before *this* commit) applied. Also, I don't see a downside in adding more flags on top, if it makes sense.

So I am going to leave this as-is for now. Could be improved in a follow-up, if needed.

Reply:
I thought about suggesting to add the WITNESS flag to the P2SH block, but decided that would just be confusing. And I'm not even sure if it's sane, because part of SegWit uses P2SH.